.github/workflows/bump-deps.yaml

```diff
@@ -0,0 +1,64 @@
+name: Bump DEPS
+on:
+  workflow_dispatch:
+    inputs:
+      dep:
+        description: "Dependency"
+        required: true
+        type: string
+        default: "vendor/web-discovery-project"
+      ref:
+        description: "New ref for dependency"
+        required: true
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: auto-bump
+        run: |
+          set -x
+
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+          git config user.name "${{ github.actor }}"
+          git config push.default simple
+
+          SED=sed
+          DATE=date
+          DEPS=DEPS
+
+          # Inputs
+          DEP="${{ inputs.dep }}"
+          DEPREF="${{ inputs.ref }}"
+          [[ -z "$DEP" ]] && { echo Input dependency cannot be empty; exit 1; }
+          [[ -z "$DEPREF" ]] && { echo Input ref cannot be empty; exit 1; }
```

Comment:
If these have default values, they're unlikely to ever be empty.

I think we can just use `${DEP:?}` and `${DEPREF:?}` the first time these variables are referenced and drop the explicit error handling. With good variable naming, the error saying that the variable is empty, or unset should be clear enough. Less code to read and maintain.

Also beware of constructs like this in bash (`condition && something`), because when used with `set -e`, it will fail the script when the condition is false. If you want to use constructs like this safely, you can use `||` instead - `! condition || something`.

Reply:
What we could do instead here is higher level validation. Make sure the DEP is only letters, hyphens and slashes and the DEPREF is only letters and numbers.

Reply:
Stricter validation done in ac83f51